vanilla/vanilla-connect-php
# Create VanillaConnect library

# Information

Original spec:
- https://github.com/vanilla/vanilla-patches/issues/93#issue-218807719
- https://github.com/vanilla/vanilla-patches/issues/99#issuecomment-301538582

Read the [README.md](https://github.com/vanilla/vanilla-connect-php/blob/b006096d5fc011c1343860aa1aa32d0ade36eea1/README.md)

Notable differences:
- We use `?jwt` instead of `?sso` since the latter is reserved to jsConnect `push sso`/`embed`
- Embed Flow === Push SSO
- The push SSO JWT' claim contains the 'aud' (audience) field which must be set to 'pushsso'

# TODO
- [x] Create the main class
    - [x] Handle request JWT
    - [x] Handle response JWT
- [x] Create a provider class for clients to use.
    - [x] URL whitelist for redirect
    - [x] Generate response URL containing either the provider's claim or the errors if something went wrong. (If the redirect url is not whitelisted we will still redirect to it but the response JWT will contain an error)
    - [x] Push SSO (allow to create a JWT that can be passed in the querystring to authenticate a user)
    - [x] Travis
- [x] README documentation
    - [x] Briefly explain the purpose of this library and how it works.
    - [x] Explain the whitelist
    - [x] Provide an example
- [x] Composer

# How to test

You can see the library used [here](https://github.com/vanilla/stub-sso-providers/pull/1/files#diff-5ff1708944bdf264dff08b3183077db2) and [here as a pushsso](https://github.com/vanilla/stub-sso-providers/pull/1/files#diff-087ead604a162f232637899477accd5e).
